### PR TITLE
Fix: Akka.Cluster Components

### PR DESCRIPTION
**Pull request by Zetanova**
Multiple components in the cluster startup like the IDowningProvider retrieved the cluster instance over `Cluster.Get(Context.System)` in the constructor. 
this is racy because the completion of the Cluster constructor extension is not guaranteed.

### CODE REVIEW

File: src/core/Akka.Cluster/ClusterDaemon.cs [hunk 1]
```diff
@@ -816,6 +816,8 @@ public ClusterEvent.IClusterDomainEvent Event
     /// </summary>
     internal sealed class ClusterDaemon : ReceiveActor, IRequiresMessageQueue<IUnboundedMessageQueueSemantics>
     {
+        private Cluster _cluster;
```
Comment (Aaronontheweb) [inline src/core/Akka.Cluster/ClusterDaemon.cs:819]: Where does this value get assigned?

**Review by Aaronontheweb — CHANGES_REQUESTED** [review]
as far as I can tell, the `ClusterDaemon` never assigns its internal `_cluster` property.

File: src/core/Akka.Cluster/ClusterDaemon.cs (hunk 1 above)
Comment (Zetanova) [reply src/core/Akka.Cluster/ClusterDaemon.cs:819]: in `ClusterDaemon .CreateChildren` method
The message `InternalClusterAction.GetClusterCoreRef` acts as the init message from `Cluster` constructor

**Review by Zetanova — COMMENTED** [review]
resolved

File: src/core/Akka.Cluster/ClusterDaemon.cs [hunk 2]
```diff
@@ -888,11 +890,12 @@ private void AddCoordinatedLeave()
             _coordShutdown.AddTask(CoordinatedShutdown.PhaseClusterShutdown, "wait-shutdown", () => _clusterPromise.Task);
         }
 
-        private void CreateChildren()
+        private void CreateChildren(Cluster cluster)
         {
+            _cluster = cluster;
```
Comment (Aaronontheweb) [inline src/core/Akka.Cluster/ClusterDaemon.cs:895]: Ok, got it

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Pull request merged**